server/src/main/java/org/elasticsearch/indices/recovery/StartRecoveryRequest.java

```diff
@@ -34,14 +35,17 @@
  */
 public class StartRecoveryRequest extends TransportRequest {
 
-    private long recoveryId;
-    private ShardId shardId;
-    private String targetAllocationId;
-    private DiscoveryNode sourceNode;
-    private DiscoveryNode targetNode;
-    private Store.MetadataSnapshot metadataSnapshot;
-    private boolean primaryRelocation;
-    private long startingSeqNo;
+    public static final Version WAIT_FOR_RELOCATION_VERSION = Version.V_8_0_0;
+
+    private final long recoveryId;
+    private final ShardId shardId;
+    private final String targetAllocationId;
+    private final DiscoveryNode sourceNode;
+    private final DiscoveryNode targetNode;
+    private final Store.MetadataSnapshot metadataSnapshot;
+    private final boolean primaryRelocation;
+    private final boolean waitIndefinitelyForRelocation;
```

Comment:
Somewhat off topic, so just to give some short justification of why I did it this way:  

I like the idea of adding this information here because we could technically use it for further efficiency gains in the future as well.
Currently, many steps in the recovery for a searchable snapshot shard seem redundant (e.g. why even send a clean files request?). Maybe we could add additional information here (or make this a more general flag) to avoid these redundant steps on the source to begin with.